src/main/java/it/gov/pagopa/pu/workflow/wf/classification/iuf/classification/IufClassificationWFImpl.java

```diff
@@ -0,0 +1,136 @@
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    IufClassificationWfConfig wfConfig = applicationContext.getBean(IufClassificationWfConfig.class);
+
+    clearClassifyIufActivity = wfConfig.buildClearClassifyIufActivityStub();
+    iufClassificationActivity = wfConfig.buildIufClassificationActivityStub();
+
+    startTransferClassificationActivity = wfConfig.buildTransferClassificationStarterHelperActivityStub();
+
+  }
+
+  @Override
+  public void classify() {
+
+    if (result != null && result.isSuccess()) {
+
+      Set<Transfer2ClassifyDTO> uniqueTransfers = new HashSet<>(result.getTransfers2classify());
+
+      // signal transfer classification for each transfer
+      uniqueTransfers.forEach(transfer2ClassifyDTO -> {
+
+        String iuv = transfer2ClassifyDTO.getIuv();
+        String iur = transfer2ClassifyDTO.getIur();
+        int transferIndex = transfer2ClassifyDTO.getTransferIndex();
+
+        startTransferClassificationActivity.signalTransferClassificationWithStart(result.getOrganizationId(), iuv, iur, transferIndex);
+      });
+
+    } else {
+      log.warn("Result is null or classification was not successful for organizationId {}",
+        result != null ? result.getOrganizationId() : "N/A");
+    }
+
+  }
+
+
+  @Override
+  public void notifyTreasury(IufClassificationNotifyTreasurySignalDTO signalDTO) {
+
+    log.info("Handling treasury notification in iuf classification: {}", signalDTO);
+    boolean clearedResult = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification cleared with result {} for {}", clearedResult, signalDTO);
+
+    IufClassificationActivityResult iufClassificationActivityResult = iufClassificationActivity.classify(signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    if (iufClassificationActivityResult.isSuccess()) {
+      if (result == null) {
+        result = iufClassificationActivityResult;
+      } else {
+        Set<Transfer2ClassifyDTO> uniqueTransfers = new HashSet<>(result.getTransfers2classify());
+        uniqueTransfers.addAll(iufClassificationActivityResult.getTransfers2classify());
+        List<Transfer2ClassifyDTO> listTransfers = uniqueTransfers.stream().toList();
+        result.setTransfers2classify(listTransfers);
+
+      }
+    } else {
+      log.error("Error in treasury classification for organizationId {}", signalDTO.getOrganizationId());
+    }
```

Comment:
just add the result (the success should be actually ignored, it will be reconsidered when refactoring exception handling)
```suggestion
    IufClassificationActivityResult iufClassificationActivityResult = iufClassificationActivity.classify(signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
    toNotify.add(iufClassificationActivityResult);
```

Reply:
OK